sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1767,7 +1767,7 @@ object GpuOverrides extends Logging {
       (a, conf, p, r) => new UnixTimeExprMeta[FromUnixTime](a, conf, p, r) {
         override def convertToGpu(lhs: Expression, rhs: Expression): GpuExpression =
           // passing the already converted strf string for a little optimization
-          GpuFromUnixTime(lhs, rhs, strfFormat)
+          GpuFromUnixTime(lhs, rhs, strfFormat, a.timeZoneId)
```

Comment:
No need to pass in ```timeZoneId```. You can directly get it since it's a ```TimeZoneAwareExpression```. See https://github.com/NVIDIA/spark-rapids/pull/9816/commits/749cfbfbab87b0722cc388371c4abcd63f0beee3#diff-9f0f07939d25080e000ca9d939ac1443eef516868843d09ccca0d44b8666d651R851

Reply:
It's a TimeZoneAwareExpression, but but we should copy the `timeZoneId` from wrapped object to here, or we'll always get None. 
By default, it's non in the constructor of `GpuFromUnixTime`
```
case class GpuFromUnixTime(
    sec: Expression,
    format: Expression,
    strfFormat: String,
    timeZoneId: Option[String] = None)   // Default it's None.
```

I think you should convert the following
```
GpuUnixTimestamp(lhs, rhs, sparkFormat, strfFormat)
```
to
```
GpuUnixTimestamp(lhs, rhs, sparkFormat, strfFormat, a. timeZoneId)
```
in your PR.